### ISSUE TITLE
content/tracing/span/links: finish section on links

### ISSUE THREAD

**Issue by odeke-em** [census-instrumentation/opencensus-website#447]
We need to update the links section https://opencensus.io/tracing/span/link/
with:
a) Code samples how to use and create links: Go, Java, C++, Python at least
b) When/how to use links
c) Visuals for what links look like -- perception makes more sense for humans to understand and apply

/cc @hvent90

**Comment by hvent90** [comment]
Posting my current progress here to field asynchronous feedback. If anyone has time, please let me know if there are errors in my API usage:

----Creating a link in Go
```go
ctxA, spanA := trace.StartSpan(context.Background(), "span A")
defer spanA.End()

ctxB, spanB := trace.StartSpan(context.Background(), "span B")
defer spanB.End()

spanAContext = spanA.SpanContext()
spanB.AddLink(trace.Link{
  TraceID: spanAContext.TraceID,
  SpanID: spanAContext.SpanID,
  Type: trace.LinkTypeUnspecified, // available types: https://github.com/census-instrumentation/opencensus-go/blob/264a2a48d94c062252389fffbc308ba555e35166/trace/basetypes.go#L72-L76
})
```

----Creating a Link in Java
```java
class MyClass {
  private static final Tracer tracer = Tracing.getTracer();
  
  void DoWork(Span parent) {
    Span spanA = tracer.spanBuilder("Span A").startSpan();
    SpanContext spanAContext = spanA.getContext();

    Span spanB = tracer.spanBuilder("Span B").startSpan();
    Link link = Link.fromSpanContext(spanAContext, Link.Type.CHILD_LINKED_SPAN);

    
    spanA.end();
    spanB.end();
  }
}
```

----Creating a Link in Python
```py
spanA = tracer.start_span(name='span A')
tracer.end_span()

spanB = tracer.start_span(name='span B')
spanB.add_link(Link(
  trace_id=spanA.context.trace_id,
  span_id=spanA.context.span_id,
  type=trace.link.Type.TYPE_UNSPECIFIED
))
tracer.end_span()
```

----Creating a Link in Node
```js
const spanA = tracer.startRootSpan('Span A');
spanA.start();

const spanB = tracer.startRootSpan('Span B');
spanB.start();
spanB.addLink(spanA.traceId, spanA.id, 'TYPE_UNSPECIFIED');

spanA.end();
spanB.end();
```

**Comment by hvent90** [comment]
Does anyone have a screenshot of what a linked span looks like in Jaeger or Zipkin? I've tried implementing links in Node & Go and exported both to Jaeger and Zipkin without seeing anything resembling a link. My eyes are untrained and may be glossing over it, though.

Code I used (links are in found highlighted lines): [Go](https://gist.github.com/hvent90/9e186b5882a3901e83fccf44cbb413e8#file-linking-go-L42-L65), [Node](https://gist.github.com/hvent90/e780c483113ca5aac44786ee183796d8#file-linking-js-L24-L40)
